### ISSUE TITLE
Unable to use date in the partitioning key

### ISSUE THREAD

**Issue by Piersen** [strapdata/elassandra#222]
**Elasticsearch version**: 6.2.3

**Plugins installed**: []

**JVM version** (`java -version`): openjdk version "1.8.0_181"

**OS version** (`uname -a` if on a Unix-like system): 05 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

**Description of the problem including expected versus actual behavior**: Unable to use date as a part of the partitioning key. Reason is stated as: "Unable to make unsigned int (for date) from: '-2147465869'". When date is removed from partitioning key in the example replicated below, the error does not occur. 

**Steps to reproduce**:

 1. In cqlsh:
```cql
CREATE KEYSPACE example WITH REPLICATION =

{ 'class':'NetworkTopologyStrategy', 'DC1':2 };

CREATE TABLE example.sessions (

  id timeuuid,

  project_id uuid,

  day date,
  
  PRIMARY KEY ((project_id, day), id)

) WITH CLUSTERING ORDER BY (id DESC);


INSERT INTO example.sessions (id, project_id, day) 
VALUES (
	now(),
	uuid(),
	toDate(now())
);
```

 2. `curl -H "Content-Type: application/json" -XPUT "http://localhost:9200/example?pretty"`
 3. `curl -XPUT "http://localhost:9200/example/_mapping/sessions" -H 'content-type: application/json' -d '{ "sessions" : { "discover" : ".*" }}'`
4. In browser, seach `http://localhost:9200/example/sessions/_search?&q=*:*&error_trace=true`

**Provide logs (if relevant)**:

**Comment by DBarthe** [comment]
Hello @Piersen,

You're right, thanks for reporting the issue. The mapping of CQL type date is buggy.

As a workaround you can use the CQL type **timestamp** rather than date, and slightly modify your insert statement with `toTimestamp(toDate(now()))`.

**Comment by vroyer** [comment]
Hi,
We have reproduced and fixed your issue, the fix will be available in the next release.
Thanks.

**Comment by vroyer** [comment]
Fixed in 6.2.3.11
